avm/src/vm/opcode.rs

```diff
@@ -3668,8 +3695,21 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
       let ns = HandlerMemory::fractal_to_string(hand_mem.read_fractal(args[0])?)?;
       let key = HandlerMemory::fractal_to_string(hand_mem.read_fractal(args[1])?)?;
       let nskey = format!("{}:{}", ns, key);
-      let ds = Arc::clone(&DS);
-      let has = ds.contains_key(&nskey);
+      let ctrl_port = CONTROL_PORT_CHANNEL.get();
+      let ctrl_port = match ctrl_port {
+        Some(ctrl_port) => Some(ctrl_port.borrow().clone()), // TODO: Use thread-local storage
+        None => None,
+      };
+      let is_key_owner = match ctrl_port {
+        Some(ref ctrl_port) => ctrl_port.is_key_owner(&nskey),
+        None => true,
+      };
+      let has = if is_key_owner {
+        let ds = Arc::clone(&DS);
+        ds.contains_key(&nskey)
+      } else {
+        ctrl_port.unwrap().dshas(&nskey).await
```

Comment:
same as above - we should probably handle the `None` case

Reply:
Same as above, the None case should be nonexistent in the future.